Content.Shared/WaterBalloon/WaterBalloonSystem.cs

```diff
@@ -0,0 +1,64 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Sprite;
+using Robust.Shared.Network;
+using Robust.Shared.Timing;
+using Robust.Shared.Containers;
+
+namespace Content.Shared.WaterBalloon;
+
+public sealed class WaterBalloonSystem : EntitySystem
+{
+    [Dependency] private readonly SharedSolutionContainerSystem _solutions = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<WaterBalloonComponent, SolutionContainerChangedEvent>(CheckIfFull);
+    }
+
+    /// <summary>
+    /// Checks if the balloon has reached its MaxVolume value 
+    /// </summary>
+    private void CheckIfFull(Entity<WaterBalloonComponent> ent, ref SolutionContainerChangedEvent args)
+    {
+        if (_gameTiming.IsFirstTimePredicted)  //prevents bad prediction shenanigans
+        {
+            if (ent.Comp.MaxVolume == args.Solution.Volume)
+            {
+                if (args.Solution.Name == null)
+                    return;
+
+                TieBalloon(ent, args.Solution, args.Solution.Name);  //send the entity solution and its name 
+            }
+        }
```

Comment:
i would combine them all in one

```suggestion
        if (_gameTiming.IsFirstTimePredicted  //prevents bad prediction shenanigans
            && ent.Comp.MaxVolume == args.Solution.Volume
            && args.Solution.Name != null)
        {
            TieBalloon(ent, args.Solution, args.Solution.Name);  //send the entity solution and its name 
        }
```

Reply:
ohhh yeah thats way more smooth